numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -621,6 +624,22 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self){
+    PyObject *arr, *ret = NULL;
+
+    if (!PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
+        return gentype_str(self);
```

Comment:
this was to maintain the old behavior. In numpy 1.13 the void scalars went though `gentype_repr` for repr and `gentype_str` for str, and both of these returned the output of `PyObject_String` (and not `PyObject_Repr`). You can see this in the diff of the 2nd commit.